pkg/virt-controller/watch/vm_test.go

```diff
@@ -5517,6 +5519,53 @@ var _ = Describe("VirtualMachine", func() {
 			Entry("networks is non-empty", []v1.Interface{}, []v1.Network{{Name: "b"}}),
 		)
 
+		DescribeTable("call to synchronizer", func(syncer testSynchronizer, expectCond v1.VirtualMachineCondition, expectCondCount int) {
+			vm, vmi := DefaultVirtualMachine(true)
+			vm, err := virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			addVirtualMachine(vm)
+			controller.Synchronizers = []synchronizer{syncer}
+
+			markAsReady(vmi)
+			vmi, err = virtFakeClient.KubevirtV1().VirtualMachineInstances(vm.Namespace).Create(context.Background(), vmi, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			controller.vmiIndexer.Add(vmi)
+
+			sanityExecute(vm)
+
+			vm, err = virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(vm.Status.Conditions).To(ContainElement(
+				gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+					"Type":    Equal(expectCond.Type),
+					"Status":  Equal(expectCond.Status),
+					"Reason":  Equal(expectCond.Reason),
+					"Message": Equal(expectCond.Message),
+				}),
+			))
+			Expect(vm.Status.Conditions).To(HaveLen(expectCondCount))
```

Comment:
Doesn't the count make this test fragile as well? Do we care how many condition are here?

Reply:
It is a balance between fragile and expressing reality.
At the moment, this expresses that there are cases where there are more conditions. For this specific test we do not care about the details of those conditions, but we acknowledge there are. If we perform a change in the codebase and this one fails, it means we may have changes something unintentionally and we should make it is ok.

Reply:
Still I think what you care about here is only the one condition. It can well be that we remove/add or do both and I can't say why you would care here.